compositor_render/src/transformations/layout.rs

```diff
@@ -143,13 +141,21 @@ impl LayoutNode {
             .collect();
         self.params.update(params, ctx.wgpu_ctx);
 
+        let textures: Vec<Option<&NodeTexture>> = layouts
+            .iter()
+            .map(|layout| match layout.content {
+                RenderLayoutContent::Color(_) => None,
+                RenderLayoutContent::ChildNode { index, .. } => Some(
+                    sources
+                        .get(index)
+                        .expect("Invalid source index in layout")
```

Comment:
Not sure if panic here would be appropriate, I would log `error!` and maybe return None